.github/workflows/test.yml

```diff
@@ -0,0 +1,30 @@
+name: Test PR — pnpm packages
```

Comment:
This is mostly copied from the private repo with a few small changes:
1. I've removed the paths checks so that the workflow runs for all paths
2. I've removed the step for uploading debug logs (mostly because I'm unsure if sensitive info can show up in these logs and what the implications of uploading them as artifacts is in a public repo)
3. I've removed the step for setting up NPM creds because they aren't needed. I'm assuming that the original workflow set up creds because some of the packages were initially private in the registry.
4. Minor naming and working dir tweaks since the dir structure has changed from the private repo